test/integration/users.test.js

```diff
@@ -1182,6 +1183,115 @@ describe("Users", function () {
+      expect(response.body.allUser).to.be.a("array");
+      response.body.allUser.forEach((status) => {
+        expect(status).to.have.property("discordJoinedAt");
+      });
+    });
+
+    it("Should get all the users with ONBOARDING state and are present in discord server for more than (given days in query)", async function () {
+      let userId1 = "";
+      let userId2 = "";
+      let userId3 = "";
+
+      userId1 = await addUser(userData[0]);
+      userId2 = await addUser(userData[1]);
+      userId3 = await addUser();
+
+      await userStatusModel.updateUserStatus(userId1, userStatusDataAfterFillingJoinSection);
+      await userStatusModel.updateUserStatus(userId2, userStatusDataAfterFillingJoinSection);
+      await userStatusModel.updateUserStatus(userId3, userStatusDataForNewUser);
+
+      const response = await chai
+        .request(app)
+        .get("/users/onboarding")
+        .query({ minPresenceDays: 20 })
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`);
+
+      expect(response).to.have.status(200);
+      expect(response.body).to.be.a("object");
+      expect(response.body.totalUsers).to.be.a("number");
+      expect(response.body.message).to.equal("All Users with ONBOARDING state of more than 20 days found successfully");
+      expect(response.body.allUser).to.be.a("array");
+      response.body.allUser.forEach((status) => {
+        expect(status).to.have.property("discordJoinedAt");
+      });
+    });
+
+    it("Should return zero users as users are not verified or discordJoinedAt property is missing", async function () {
+      let userId1 = "";
+      let userId2 = "";
+
+      userId1 = await addUser(userData[2]);
+      userId2 = await addUser(userData[3]);
+
+      await userStatusModel.updateUserStatus(userId1, userStatusDataAfterFillingJoinSection);
+      await userStatusModel.updateUserStatus(userId2, userStatusDataAfterFillingJoinSection);
+
+      const response = await chai
+        .request(app)
+        .get("/users/onboarding")
+        .query({ minPresenceDays: 30 })
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`);
+
+      expect(response).to.have.status(200);
+      expect(response.body.allUser.length).to.equal(0);
+      expect(response.body.totalUsers).to.equal(0);
+      expect(response.body.message).to.equal(
+        `No users exist with ONBOARDING state of more than 30 days, or the user has not been verified`
+      );
+    });
+
+    it("Should throw for no user is with ONBOARDING state", async function () {
+      let userId1 = "";
+      let userId2 = "";
+      let userId3 = "";
+
+      userId1 = await addUser(userData[0]);
+      userId2 = await addUser(userData[1]);
+      userId3 = await addUser();
+
+      await userStatusModel.updateUserStatus(userId1, userStatusDataForNewUser);
+      await userStatusModel.updateUserStatus(userId2, userStatusDataForNewUser);
+      await userStatusModel.updateUserStatus(userId3, userStatusDataForNewUser);
```

Comment:
I see in all test you are repeating this line from 1273 to 1283...can we put this in beforeEach??
Correct me if i am wrong??

Reply:
No, as both user data and user status data are different for each test. 
At some test I'm using `userStatusDataAfterFillingJoinSection` status data but in some place `userStatusDataForNewUser` is being used. as both are different.